### ISSUE TITLE
M1 suppot

### ISSUE THREAD

**Issue by wzh4464** [mathworks/MATLAB-extension-for-vscode#53]
**Describe the bug**
Error in finding `mlint`

**To Reproduce**
1. Use an M1 apple silicon computer
2. activate the mlint procedure

**Expected behavior**
`Error: spawn /Applications/MATLAB_R2023b.app/bin/maci64/mlint ENOENT`

**Screenshots**
<img width="742" alt="image" src="https://github.com/mathworks/MATLAB-extension-for-vscode/assets/32484940/27fc5a4b-3973-4663-b8ec-fd526035514e">

**Useful Information**
 - OS Version: MacOS Ventura
 - VS Code Version: 

**Additional context**
The extension just require the matlabroot but fail to recognize its arm arch.

**Comment by dklilley** [comment]
Thanks for reporting this!

I unfortunately do not have access to a M1 Mac with MATLAB installed right now. Can you share the expected path were the mlint executable can be found? Is it under `maca64` instead of `maci64`?

I believe this issue should be limited to linting when mlint is being used. Can you confirm whether you are experiencing any other issues with the extension?

**Comment by dklilley** [comment]
Thanks for the confirmation!

**Comment by shanemcq18** [comment]
Same problem here on an M2 Apple silicon laptop, but I'm also wondering if this may also be preventing MATLAB from connecting to the extension. So far I haven't been able to make a connection without an immediate error/disconnect.

Version info:
- MATLAB `R2023b` (Apple silicon version)
- Visual Studio Code `1.83.1`
- MATLAB extension for Visual Studio Code `v1.1.4`.
- `/Applications/MATLAB_R2023b.app/bin/` contains a `maca64/` folder but not a `maci64/` folder.
- `/Applications/MATLAB_R2023b.app/bin/maca64/` contains `mlint` and a `matlabconnector/` folder (not sure if that's related to the issue, but it seems like the root problem is that `maca64/` isn't being found).

MATLAB Language Server OUTPUT log:
```text
[Error - 10:06:29 AM] (10:06:29) matlabls - ERROR: Error from mlint executable: spawn /Applications/MATLAB_R2023b.app/bin/maci64/mlint ENOENT
Error: spawn /Applications/MATLAB_R2023b.app/bin/maci64/mlint ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
```
This is where it fails to find `mlint`. Now if we try to connect to MATLAB:

```text
(10:09:30) matlabls: Launching MATLAB...
(10:10:31) matlabls: Started MATLAB
[Warn  - 10:10:31 AM] (10:10:31) matlabls - WARNING: Error while communicating with MATLAB - disconnecting
```
So technically it connects, but it immediately disconnects every time. Is that because of failing to find `maca64/`? Not sure how to debug this further.

This is actually a [MATLAB-language-server](https://github.com/mathworks/MATLAB-language-server) (`matlabls`) issue, perhaps originating in [`src/providers/linting/LintingSupportProvider.ts`](https://github.com/mathworks/MATLAB-language-server/blob/9fdc85418b16f98850fc455116dada3bd51921c4/src/providers/linting/LintingSupportProvider.ts#L372)?

```typescript
    /**
     * Gets the name of platform-specific binary directory.
     *
     * @returns The binary directory name, or null if the platform is not recognized
     */
    private getBinDirectoryForPlatform (): string | null {
        switch (process.platform) {
            case 'win32':
                return 'win64'
            case 'darwin':
                return 'maci64'          // <-- need to check if silicon or not?
            case 'linux':
                return 'glnxa64'
            default:
                return null
        }
    }
```

Thanks for the extension, it's nice to be able to leverage Visual Studio Code directly for MATLAB code development.

**Comment by dklilley** [comment]
Hi @shanemcq18 , I don't think the problem captured in this issue would prevent you from connecting to MATLAB, so my suspicion is that there may be something else going wrong there. 

My first guess is that there may be a proxy issue in play. Please try the steps hamidingit suggested in his first comment on [this issue](https://github.com/mathworks/MATLAB-language-server/issues/10). The output of the `curl` command should indicate whether there is a proxy blocking the connection. If, however, the output looks similar to krzysid's response, this is not the case, and I would suggest reaching out to support@mathworks.com so we can try to look into this in more detail.

**Comment by shanemcq18** [comment]
@dklilley thanks for the response. Quick followup, here's my `curl` output (port 53408):

```text
* Uses proxy env variable no_proxy == '127.0.0.1, localhost, <some others for my company>
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
*   Trying 127.0.0.1:53408...
* Connected to 127.0.0.1 (127.0.0.1) port 53408 (#0)
* ALPN: offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
} [296 bytes data]
* (304) (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
* (304) (IN), TLS handshake, Unknown (8):
{ [22 bytes data]
* (304) (IN), TLS handshake, Certificate (11):
{ [870 bytes data]
* (304) (IN), TLS handshake, CERT verify (15):
{ [264 bytes data]
* (304) (IN), TLS handshake, Finished (20):
{ [52 bytes data]
* (304) (OUT), TLS handshake, Finished (20):
} [52 bytes data]
* SSL connection using TLSv1.3 / AEAD-AES256-GCM-SHA384
* ALPN: server did not agree on a protocol. Uses default.
* Server certificate:
*  subject: C=US; O=company; CN=127.0.0.1; OU=engineering
*  start date: Nov  7 17:01:18 2023 GMT
*  expire date: Nov  6 17:01:18 2024 GMT
*  issuer: C=US; O=company; CN=127.0.0.1; OU=engineering
*  SSL certificate verify result: self signed certificate (18), continuing anyway.
* using HTTP/1.x
> GET / HTTP/1.1
> Host: 127.0.0.1:53408
> User-Agent: curl/8.1.2
> Accept: */*
> 
< HTTP/1.1 200 OK
< Date: Tue, 07 Nov 2023 17:03:11 GMT
< Connection: Keep-Alive
< Pragma: no-cache
< X-Content-Security-Policy: default-src 'self' *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:*; style-src 'self' 'unsafe-inline' *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:*; script-src 'self' 'unsafe-inline' 'unsafe-eval' *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:*; img-src 'self' blob *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:* data:; font-src 'self' *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:* data:; frame-ancestors 'self' *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:*; frame-src 'self' https://www.getfeedback.com *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:*; connect-src 'self' blob: *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:* wss://localhost:* wss://127.0.0.1:*
< Strict-Transport-Security: max-age=31536000; includeSubDomains
< Cache-Control: no-cache, no-store, must-revalidate
< X-XSS-Protection: 1; mode=block
< Expires: Wed, 21 Oct 2015 07:28:00 GMT
< Referrer-Policy: strict-origin
< Content-Length: 211
< X-Content-Type-Options: nosniff
< ETag: "1459445420211"
< Last-Modified: Thu, 31 Mar 2016 17:30:20 GMT
< Content-Type: text/html
< Accept-Ranges: bytes
< X-Frame-Options: sameorigin
< Content-Security-Policy: default-src 'self' *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:*; style-src 'self' 'unsafe-inline' *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:*; script-src 'self' 'unsafe-inline' 'unsafe-eval' *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:*; img-src 'self' blob *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:* data:; font-src 'self' *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:* data:; frame-ancestors 'self' *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:*; frame-src 'self' https://www.getfeedback.com *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:*; connect-src 'self' blob: *.mathworks.com:* *.thingspeak.com:* thingspeak.com:* *.matlab.com:* https://localhost:* https://127.0.0.1:* wss://localhost:* wss://127.0.0.1:*
< Set-Cookie: JSESSIONID2=EX6uGRFMUnDhJcGU; Path=/; HttpOnly; Secure; SameSite=None
< 
{ [211 bytes data]

100   211  100   211    0     0   6628      0 --:--:-- --:--:-- --:--:--  7814
* Connection #0 to host 127.0.0.1 left intact
<!doctype html>
<html>
<head>
    <title>connector service</title>
	<link href="/css/connector.css" rel="stylesheet" type="text/css" />
</head>
<body>
	<h1>The connector service is running.</h1>
</body>
</html>
```

The top block (lines with `>`) look similar to the output in the issue you mentioned. But the next block (lines with `<`) looks a little different. I don't know enough here to tell if it's a proxy issue or not, but that would make sense as my workplace requires a few proxies. Any ideas on how to get the extension to talk to the proxies? No worries if this is out of scope of the original issue and the one you referenced.

**Comment by dklilley** [comment]
@shanemcq18 , we would like to look into this further. Can you reach out to support@mathworks.com and reference this issue? This will allow us to investigate further and avoid sharing more sensitive log files over GitHub.

**Comment by dklilley** [comment]
This has been resolved in v1.1.5 (#77)
